types/setting.go

```diff
@@ -940,30 +972,47 @@ func ValidateSetting(name, value string) (err error) {
 	case SettingNameBackupstorePollInterval:
 		interval, err := strconv.Atoi(value)
 		if err != nil {
-			return fmt.Errorf("value is not int: %v", err)
+			errors.Wrapf(err, "value %v is not a number", value)
 		}
 		if interval < 0 {
 			return fmt.Errorf("the value %v shouldn't be less than 0", value)
 		}
 	case SettingNameFailedBackupTTL:
 		interval, err := strconv.Atoi(value)
 		if err != nil {
-			return fmt.Errorf("value is not int: %v", err)
+			errors.Wrapf(err, "value %v is not a number", value)
 		}
 		if interval < 0 {
 			return fmt.Errorf("the value %v shouldn't be less than 0", value)
 		}
 	case SettingNameTaintToleration:
 		if _, err = UnmarshalTolerations(value); err != nil {
-			return fmt.Errorf("the value of %v is invalid: %v", sName, err)
+			return errors.Wrapf(err, "the value of %v is invalid", sName)
 		}
 	case SettingNameSystemManagedComponentsNodeSelector:
 		if _, err = UnmarshalNodeSelector(value); err != nil {
-			return fmt.Errorf("the value of %v is invalid: %v", sName, err)
+			return errors.Wrapf(err, "the value of %v is invalid", sName)
 		}
 	case SettingNameStorageNetwork:
 		if err = ValidateStorageNetwork(value); err != nil {
-			return fmt.Errorf("the value of %v is invalid: %v", sName, err)
+			return errors.Wrapf(err, "the value of %v is invalid", sName)
+		}
+
+	case SettingNameRecurringSuccessfulJobsHistoryLimit:
+		successJobsHistLimit, err := strconv.Atoi(value)
+		if err != nil {
+			return errors.Wrapf(err, "value %v is not a number", value)
+		}
+		if successJobsHistLimit < 0 {
+			return fmt.Errorf("the value %v shouldn't be less than 0", value)
+		}
+	case SettingNameRecurringFailedJobsHistoryLimit:
+		failedJobsHistLimit, err := strconv.Atoi(value)
+		if err != nil {
+			return errors.Wrapf(err, "value %v is not a number", value)
+		}
+		if failedJobsHistLimit < 0 {
+			return fmt.Errorf("the value %v shouldn't be less than 0", value)
```

Comment:
What does 0 mean in the two settings? If it's meaningful, you can mention in the setting description.

Reply:
Set the limit to 0 means that it won't keep history after the job finished. And I will mention it in the description.

https://github.com/longhorn/longhorn-manager/blob/35d60930fc09d0b4d37218488aa11dcdb0e91942/types/setting.go#L827-L828
https://github.com/longhorn/longhorn-manager/blob/35d60930fc09d0b4d37218488aa11dcdb0e91942/types/setting.go#L838-L839